### PR TITLE
Make ADIOS tests compatible with 2.9 and 2.10

### PR DESCRIPTION
**Pull request by jhale**
We know we are not compatible with 2.10. Will do a minor DOLFINx release once ADIOS 2.10 is widely available.

### CODE REVIEW

**Comment by jorgensd** [comment]
> We know we are not compatible with 2.10. Will do a minor DOLFINx release once ADIOS 2.10 is widely available.

I'll add a try except in the test, give me a few minutes to build adios 2.10 locally to make a fix

**Comment by jorgensd** [comment]
@jhale I added a patch to the test. As the only thing that doesn't work with adios2 is the test, i think it is better to patch the test for backwards compatibility than restricting build versions

**Comment by garth-wells** [comment]
Why not apply fixes to be compatible with 2.10? How much work would it be?

File: python/test/unit/io/test_adios2.py [hunk 1]
```diff
@@ -313,7 +313,14 @@ def test_vtx_reuse_mesh(tempdir, dim, simplex, reuse):
...
     )  # For mesh variables the step count is 1 if reuse else number of writes
 
-    adios_file = adios2.open(str(filename), "r", comm=mesh.comm, engine_type="BP4")
+    # backwards compatibility adios2 < 2.10.0
+    try:
+        adios_file = adios2.open(str(filename), "r", comm=mesh.comm, engine_type="BP4")
+    except AttributeError:
+        adios = adios2.Adios(comm=mesh.comm)
+        io = adios.declare_io("TestData")
+        adios_file = adios2.Stream(io, str(filename), "r", mesh.comm)
```
Comment (francesco-ballarin) [inline python/test/unit/io/test_adios2.py:322]: So with 2.10 one can't say anymore that `engine_type="BP4"`? Or is the default?

**Review by francesco-ballarin — APPROVED** [review]

**Comment by jorgensd** [comment]
> Why not apply fixes to be compatible with 2.10? How much work would it be?

Ive changed the scope of the PR. We are now compatible with 3.10, given that try except clause in the adios tests (only Python interface changed from 2.9 to 2.10).

**Review by garth-wells — APPROVED** [review]

File: python/test/unit/io/test_adios2.py [hunk 2]
```diff
@@ -313,7 +313,15 @@ def test_vtx_reuse_mesh(tempdir, dim, simplex, reuse):
         1 if reuse else 3
     )  # For mesh variables the step count is 1 if reuse else number of writes
 
-    adios_file = adios2.open(str(filename), "r", comm=mesh.comm, engine_type="BP4")
+    # backwards compatibility adios2 < 2.10.0
```
Comment (garth-wells) [inline python/test/unit/io/test_adios2.py:316]: Could you make clear with a comment which part of the try/except clause is for v2.9 and which is for v2.10?

File: python/test/unit/io/test_adios2.py [hunk 3]
```diff
@@ -313,7 +313,15 @@ def test_vtx_reuse_mesh(tempdir, dim, simplex, reuse):
...
+    try:
+        adios_file = adios2.open(str(filename), "r", comm=mesh.comm, engine_type="BP4")
+    except AttributeError:
+        adios = adios2.Adios(comm=mesh.comm)
```
Comment (jorgensd) [inline python/test/unit/io/test_adios2.py:320]:
```suggestion
        # adios2 >= v2.10.0
        adios = adios2.Adios(comm=mesh.comm)
```

**Pull request merged**